core/types/block.libevm.go

```diff
@@ -108,3 +109,93 @@ func (*NOOPHeaderHooks) DecodeRLP(h *Header, s *rlp.Stream) error {
 	type withoutMethods Header
 	return s.Decode((*withoutMethods)(h))
 }
+
+var (
+	_ interface {
+		rlp.Encoder
+		rlp.Decoder
+	} = (*Body)(nil)
+
+	// The implementations of [Body.EncodeRLP] and [Body.DecodeRLP] make
+	// assumptions about the struct fields, which we lock in here as a change
+	// detector. If this breaks then it MUST be updated and the RLP methods
+	// reviewed + new backwards-compatibility tests added.
+	_ = &Body{[]*Transaction{}, []*Header{}, []*Withdrawal{}}
+)
+
+// EncodeRLP implements the [rlp.Encoder] interface.
+func (b *Body) EncodeRLP(dst io.Writer) error {
+	w := rlp.NewEncoderBuffer(dst)
+
+	return w.InList(func() error {
+		if err := rlp.EncodeListToBuffer(w, b.Transactions); err != nil {
+			return err
+		}
+		if err := rlp.EncodeListToBuffer(w, b.Uncles); err != nil {
+			return err
+		}
+
+		hasLaterOptionalField := b.Withdrawals != nil
+		if err := b.hooks().AppendRLPFields(w, hasLaterOptionalField); err != nil {
+			return err
+		}
+		if !hasLaterOptionalField {
+			return nil
+		}
+		return rlp.EncodeListToBuffer(w, b.Withdrawals)
+	})
+}
+
+// DecodeRLP implements the [rlp.Decoder] interface.
+func (b *Body) DecodeRLP(s *rlp.Stream) error {
+	return s.FromList(func() error {
+		txs, err := rlp.DecodeList[Transaction](s)
+		if err != nil {
+			return err
+		}
+		uncles, err := rlp.DecodeList[Header](s)
+		if err != nil {
+			return err
+		}
+		*b = Body{
+			Transactions: txs,
+			Uncles:       uncles,
+		}
+
+		if err := b.hooks().DecodeExtraRLPFields(s); err != nil {
+			return err
+		}
+		if !s.MoreDataInList() {
+			return nil
+		}
+
+		ws, err := rlp.DecodeList[Withdrawal](s)
+		if err != nil {
+			return err
+		}
+		b.Withdrawals = ws
+		return nil
+	})
+}
+
+type BodyHooks interface {
+	AppendRLPFields(_ rlp.EncoderBuffer, mustWriteEmptyOptional bool) error
+	DecodeExtraRLPFields(*rlp.Stream) error
+}
+
+func TestOnlyRegisterBodyHooks(h BodyHooks) {
+	testonly.OrPanic(func() {
+		todoRegisteredBodyHooks = h
+	})
+}
+
+var todoRegisteredBodyHooks BodyHooks = NOOPBodyHooks{}
```

Comment:
nit should this be perhaps renamed to just `noopBodyHooks` instead of `todoRegisteredBodyHooks`? Not too clear on what's "TODO" 🤔

Reply:
It's been commented now. The TODO is the fact that it's being replaced with `RegisterExtras`.